src/ResourceManagement/Locks/ManagementLocksImpl.cs

```diff
@@ -0,0 +1,299 @@
+        IManagementLocks
+    {
+        private IAuthorizationManager manager;
+
+                ///GENMHASH:836797384EAD11F4F592C5C904884C2A:D544D70B59C64F2C67EC02DE16CBAEF6
+        internal ManagementLocksImpl(IAuthorizationManager manager)
+        {
+            this.manager = manager;
+        }
+
+        IManagementLocksOperations IHasInner<IManagementLocksOperations>.Inner
+        {
+            get
+            {
+                return Inner();
+            }
+        }
+
+        ///GENMHASH:B6961E0C7CB3A9659DE0E1489F44A936:168EFDB95EECDB98D4BDFCCA32101AC1
+        public IAuthorizationManager Manager()
+        {
+            return manager;
+        }
+
+        #region Get
+
+        ///GENMHASH:7D93B4EC99C64989F97B3D17F88C3F2C:45E96D80A053DE48AD3DD26FC5CD83FF
+        public IManagementLock GetByResourceGroup(string resourceGroupName, string name)
+        {
+            return Extensions.Synchronize(() => GetByResourceGroupAsync(resourceGroupName, name));
+        }
+
+        ///GENMHASH:5002116800CBAC02BBC1B4BF62BC4942:E5DB8B81D366BCF6A1B75F4B1D85B1F1
+        public IManagementLock GetById(string id)
+        {
+            return Extensions.Synchronize(() => GetByIdAsync(id));
+        }
+
+        ///GENMHASH:E776888E46F8A3FC56D24DF4A74E5B74:FC67528805BD588653B69DE503549F4C
+        public async Task<IManagementLock> GetByIdAsync(string id, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            string resourceId = ResourceIdFromLockId(id);
+            string lockName = ResourceUtils.NameFromResourceId(id);
+            IManagementLock l = null;
+            if (resourceId != null && lockName != null)
+            {
+                l = WrapModel(await Inner().GetByScopeAsync(resourceId, lockName));
+            }
+            return l;
+        }
+
+        ///GENMHASH:2B19D171A02814189E0329A91320316B:E509DF22AA2F28442E1D34ACD9FCD647
+        public async Task<IManagementLock> GetByResourceGroupAsync(string resourceGroupName, string name, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            return WrapModel(await Inner().GetAtResourceGroupLevelAsync(resourceGroupName, name, cancellationToken));
+        }
+
+        #endregion
+
+        #region Delete
+
+        public override void DeleteById(string id)
+        {
+            Extensions.Synchronize(() => DeleteByIdAsync(id));
+        }
+
+        ///GENMHASH:4D33A73A344E127F784620E76B686786:DF0ACE79B69D4230630060F3C796E8A9
+        public async override Task DeleteByIdAsync(string id, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            string scope = ResourceIdFromLockId(id);
+            string lockName = ResourceUtils.NameFromResourceId(id);
+            if (scope != null && lockName != null)
+            {
+                await Inner().DeleteByScopeAsync(scope, lockName);
```

Comment:
>await Inner().DeleteByScopeAsync(scope, lockName); [](start = 16, length = 50)

this one as well -  please pass cancellationToken to the call

Reply:
Doh!!
(good eye, good eye...  and I did look for these suckers...)